vm/inc/ubpf.h

```diff
@@ -42,227 +46,265 @@ extern "C" {
- * for implementing function like the "bpf_tail_call" helper.
- *
- * @param[in] vm The VM to set the unwind helper in.
- * @param[in] idx Index of the helper function to unwind on success.
- * @retval 0 Success.
- * @retval -1 Failure.
- */
-int
-ubpf_set_unwind_function_index(struct ubpf_vm* vm, unsigned int idx);
+    /**
+     * @brief Instruct the uBPF runtime to apply unwind-on-success semantics to a helper function.
+     * If the function returns 0, the uBPF runtime will end execution of
+     * the eBPF program and immediately return control to the caller. This is used
+     * for implementing function like the "bpf_tail_call" helper.
+     *
+     * @param[in] vm The VM to set the unwind helper in.
+     * @param[in] idx Index of the helper function to unwind on success.
+     * @retval 0 Success.
+     * @retval -1 Failure.
+     */
+    int
+    ubpf_set_unwind_function_index(struct ubpf_vm* vm, unsigned int idx);
 
-/**
- * @brief Override the storage location for the BPF registers in the VM.
- *
- * @param[in] vm The VM to set the register storage in.
- * @param[in] regs The register storage.
- */
-void
-ubpf_set_registers(struct ubpf_vm* vm, uint64_t* regs);
+    /**
+     * @brief Override the storage location for the BPF registers in the VM.
+     *
+     * @param[in] vm The VM to set the register storage in.
+     * @param[in] regs The register storage.
+     */
+    void
+    ubpf_set_registers(struct ubpf_vm* vm, uint64_t* regs);
 
-/**
- * @brief Retrieve the storage location for the BPF registers in the VM.
- *
- * @param[in] vm The VM to get the register storage from.
- * @return uint64_t* A pointer to the register storage.
- */
-uint64_t*
-ubpf_get_registers(const struct ubpf_vm* vm);
+    /**
+     * @brief Retrieve the storage location for the BPF registers in the VM.
+     *
+     * @param[in] vm The VM to get the register storage from.
+     * @return uint64_t* A pointer to the register storage.
+     */
+    uint64_t*
+    ubpf_get_registers(const struct ubpf_vm* vm);
 
-/**
- * @brief Optional secret to improve ROP protection.
- *
- * @param[in] vm The VM to set the secret for.
- * @param[in] secret Optional secret to improve ROP protection.
- * Returns 0 on success, -1 on error (e.g. if the secret is set after
- * the instructions are loaded).
- */
-int
-ubpf_set_pointer_secret(struct ubpf_vm* vm, uint64_t secret);
+    /**
+     * @brief Optional secret to improve ROP protection.
+     *
+     * @param[in] vm The VM to set the secret for.
+     * @param[in] secret Optional secret to improve ROP protection.
+     * Returns 0 on success, -1 on error (e.g. if the secret is set after
+     * the instructions are loaded).
+     */
+    int
+    ubpf_set_pointer_secret(struct ubpf_vm* vm, uint64_t secret);
+
+    /**
+     * @brief Data relocation function that is called by the VM when it encounters a
+     * R_BPF_64_64 relocation in the maps section of the ELF file.
```

Comment:
Do we want to be this specific in the documentation? It may be that the use case is completely orthogonal to maps. For instance, what I am using it for is independent of maps (it just targets some data). (see below)